### DISCUSSION TITLE
Retrieving host information (LISP)

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#125] [Q&A] [answered]
I would like to create a LISP that will load the appropriate `RxLoader...` file, for this I need to detect what host application I am dealing with and what version. Any ideas? I don't know much about lisp.

**Comment by CEXT-Dan** [comment]
I don’t know if there is a reliable way across all platforms, In AutoCAD, I would use something like 
```
(setq acad_ver (strcat (getvar 'product) "-" (substr (getvar "ACADVER") 1 4))); = "AutoCAD-25.0"
```
I don’t see that it works well in zwcad. Maybe parse (vlax-product-key)

**Reply by CEXT-Dan** [reply]
I asked here  https://www.theswamp.org/index.php?topic=59836

**Reply by CEXT-Dan** [reply, marked as answer]
Lee (https://lee-mac.com/) at theswamp gave a good hint 
https://www.theswamp.org/index.php?topic=59836.msg622582#msg622582

```(strcat (getvar 'product) "-" (substr (vla-get-version (vlax-get-acad-object)) 1 4))```

>"BricsCAD-25.0"
"BricsCAD-24.0"
"GstarCAD-25.0
"GstarCAD-24.0"
"AutoCAD-24.0" 
"AutoCAD-24.3"
"AutoCAD-25.0"
"ZWCAD-2024"
"ZWCAD-2025"

**Reply by gswifort** [reply]
Thanks!

**Comment by gswifort** [comment]
Do you have access to all four programs? Autocad, zwcad, etc? Do you have paid licenses, is there a way to get licenses for testing purposes?

**Reply by CEXT-Dan** [reply]
Yes, I applied to be a developer for each application, each have their own requirements and costs. That’s the only way I know aside from using the trial versions.

**Comment by CEXT-Dan** [comment]
FYI, JasonB created a loader for autolisp here
https://www.theswamp.org/index.php?topic=59872
he's a lisp god, so probably better than what I could write!

**Reply by gswifort** [reply]
Thanks!
